static/src/javascripts/projects/common/modules/crosswords/hidden-input.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import { createClass, createElement, findDOMNode } from 'react/addons';
+import bonzo from 'bonzo';
+import fastdom from 'fastdom';
+import { scrollTo } from 'lib/scroller';
+import { isBreakpoint } from 'lib/detect';
+
+const HiddenInput = createClass({
+    getInitialState() {
+        return {
+            value: this.props.value,
+        };
+    },
+
+    componentDidUpdate() {
+        if (
+            isBreakpoint({
+                max: 'mobile',
+            })
+        ) {
+            fastdom.read(() => {
+                const offsets = bonzo(findDOMNode(this.refs.input)).offset();
+                const clueHeight = document.getElementsByClassName(
+                    'crossword__sticky-clue'
+                )[0].offsetHeight;
+
+                scrollTo(
+                    offsets.top - offsets.height * 1.5 - clueHeight,
+                    250,
+                    'easeOutQuad'
+                );
+            });
+        }
+    },
+
+    handleChange(event) {
```

Comment:
It would be useful to have type annotations for the function signatures in this class 🙏

Reply:
I was going to ask, what would you recommend I do for event types? I guess flow doesn't have builtins for this?

Reply:
Oh wait, it does

Reply:
Yeah, I'd probably go with [this](https://github.com/facebook/flow/blob/v0.56.0/lib/dom.js#L213). If you haven't already seen it, the [Flow Cheat Sheet](https://www.saltycrane.com/flow-type-cheat-sheet/latest/) is great for discovering what types to use

Reply:
I ran into problems using Event because flow doesn't like event.target.value https://github.com/facebook/flow/issues/218#issuecomment-259780131

Reply:
+ https://flow.org/en/docs/react/events/